### ISSUE TITLE
ArgumentError comparing a BigDecimal to a Float

### ISSUE THREAD

**Issue by findchris** [oracle/truffleruby#1744]
As posted on [gitter](https://gitter.im/graalvm/truffleruby), I'm comparing a `BigDecimal` to a `Float`. No problems on MRI, error on TruffleRuby: 
```ruby
sum = BigDecimal("0")
10_000.times do
  sum = sum + BigDecimal("0.0001")
end
[sum, 1.1].min
# => #<BigDecimal:7f98c8868338,'0.1E1',9(18)> (MRI)
# ArgumentError: comparison of Float with 1.0 failed (TruffleRuby)
```

**Comment by rafaelfranca** [comment]
This can be closed now. It works fine on master:

```
>> require 'bigdecimal'
=> true
>> sum = BigDecimal("0")
=> 0.0
>> 10_000.times do
?>       sum = sum + BigDecimal("0.0001")
>>   end
=> 10000
>> [sum, 1.1].min
=> 0.1e1
```

**Comment by eregon** [comment]
I verified, it's fixed in 20.0.0, but not in 19.3.0.
